src/darwin/Framework/CHIP/MTRAsyncCallbackWorkQueue.h

```diff
@@ -23,6 +23,16 @@ NS_ASSUME_NONNULL_BEGIN
 
 typedef void (^MTRAsyncCallbackReadyHandler)(id context, NSUInteger retryCount);
 
+// The batching handler is called by the work queue when a work item is batchable. The handler will be passed the opaque data from
+// the current and the next work item, and should the next item's data be fully merged into the first, the fullyMerged BOOL should
+// be set to YES, so that the work queue can remove the "next item" from the queue. And when fullyMerged is set to YES, this handler
+// will be called again, if the following item is also batchable with the same ID.
```

Comment:
> That said, why is this using an outparam instead of just returning BOOL? Just for readability?

Yes that's my reason. I actually first tried returning BOOL but couldn't find a good naming scheme that clearly conveys it.